sql/src/test/java/org/apache/druid/sql/calcite/schema/InformationSchemaTest.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.type.SqlTypeFamily;
+import org.apache.druid.segment.column.ColumnType;
+import org.apache.druid.sql.calcite.BaseCalciteQueryTest;
+import org.apache.druid.sql.calcite.expression.DirectOperatorConversion;
+import org.apache.druid.sql.calcite.expression.OperatorConversions;
+import org.apache.druid.sql.calcite.expression.SqlOperatorConversion;
+import org.apache.druid.sql.calcite.planner.DruidOperatorTable;
+import org.apache.druid.sql.calcite.planner.PlannerConfig;
+import org.apache.druid.sql.calcite.table.RowSignatures;
+import org.apache.druid.sql.calcite.util.CalciteTests;
+import org.apache.druid.sql.calcite.util.QueryFrameworkUtils;
+import org.apache.druid.sql.calcite.util.SqlTestFramework;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class InformationSchemaTest extends BaseCalciteQueryTest
+{
+  private InformationSchema informationSchema;
+  private SqlTestFramework qf;
+
+  @Before
+  public void setUp()
+  {
+    qf = queryFramework();
+    DruidSchemaCatalog rootSchema = QueryFrameworkUtils.createMockRootSchema(
+        CalciteTests.INJECTOR,
+        qf.conglomerate(),
+        qf.walker(),
+        new PlannerConfig(),
+        null,
+        new NoopDruidSchemaManager(),
+        CalciteTests.TEST_AUTHORIZER_MAPPER
+    );
+
+    informationSchema = new InformationSchema(
+        rootSchema,
+        CalciteTests.TEST_AUTHORIZER_MAPPER,
+        qf.operatorTable()
+    );
+  }
+
+  @Test
+  public void testGetTableNamesMap()
+  {
+    Assert.assertEquals(
+        ImmutableSet.of("SCHEMATA", "TABLES", "COLUMNS", "ROUTINES"),
+        informationSchema.getTableNames()
+    );
+  }
+
+  @Test
+  public void testScanRoutinesTable()
+  {
+    DruidOperatorTable operatorTable = qf.operatorTable();
+    InformationSchema.RoutinesTable routinesTable = new InformationSchema.RoutinesTable(operatorTable);
+    DataContext dataContext = createDataContext();
+
+    List<Object[]> rows = routinesTable.scan(dataContext).toList();
+
+    Assert.assertTrue("There should be at least 10 functions", rows.size() > 10);
+    RelDataType rowType = routinesTable.getRowType(new JavaTypeFactoryImpl());
+    Assert.assertEquals(6, rowType.getFieldCount());
+
+    for (Object[] row : rows) {
+      Assert.assertEquals(rowType.getFieldCount(), row.length);
+      Assert.assertEquals("druid", row[0]);
+      Assert.assertEquals("INFORMATION_SCHEMA", row[1]);
+      Assert.assertNotNull(row[2]);
+      Assert.assertNotNull(row[3]);
+      String isAggregator = row[4].toString();
+      Assert.assertTrue(isAggregator.contains("YES") || isAggregator.contains("NO"));
+      // nothing to validate for signatures as it may be not be present if operandTypeChecker is not defined.
```

Comment:
Should we validate this field for one or a few known functions, that do have the operandTypeChecker defined, either in this test or another?

Reply:
ah nvm, I see you have that in the test below, nice!